### PR TITLE
Refactor normalize quotes

### PR DESCRIPTION
**Pull request by lauterb**
Added a test for NormalizeMicrosoftWordQuotes, then refactored the code a bit to use a Dictionary

### CODE REVIEW

File: CorpusSearch/Extensions/NormalizationExtensions.cs [hunk 1]
```diff
@@ -29,36 +30,43 @@ public static string RemoveDiacritics(this string target)
...
+            { '\u201c', "\"" },
+            { '\u201d', "\"" },
+            { '\u201e', "\"" },
+            { '\u2026', "..." },
+            { '\u2032', "\'" },
+            { '\u2033', "\"" }
+        };
+        foreach (var key in qmap.Keys)
+            if (buffer.IndexOf(key) > -1)
+                buffer = buffer.Replace(key.ToString(), qmap[key]);
```
Comment (david-allison) [inline CorpusSearch/Extensions/NormalizationExtensions.cs:53]: This is slower, as you're using Replace(string, string) Vs Replace(char, char)

Is there a practical impact on this on startup?

File: CorpusSearch/Extensions/NormalizationExtensions.cs [hunk 2]
```diff
@@ -29,36 +30,43 @@ public static string RemoveDiacritics(this string target)
...
         return buffer;
-    }
+        }
 
-    public static string RemoveDoubleQuotes(this string target)
-    {
-        return target.Replace("\"", "");
-    }
+        public static string RemoveDoubleQuotes(this string target)
```
Comment (david-allison) [inline CorpusSearch/Extensions/NormalizationExtensions.cs:57]: Revert spacing changes

File: CorpusSearch/Extensions/NormalizationExtensions.cs (hunk 1 above)
Comment (lauterb) [reply CorpusSearch/Extensions/NormalizationExtensions.cs:53]: I can go back to replace char and isolate the one that is a string

File: CorpusSearch/Extensions/NormalizationExtensions.cs (hunk 2 above)
Comment (lauterb) [reply CorpusSearch/Extensions/NormalizationExtensions.cs]: got it

File: CorpusSearch/Extensions/NormalizationExtensions.cs [hunk 3]
```diff
@@ -29,20 +30,28 @@ public static string RemoveDiacritics(this string target)
...
+            { '\u201a', ',' },
+            { '\u201b', '\'' },
+            { '\u201c', '\"' },
+            { '\u201d', '\"' },
+            { '\u201e', '\"' },
+            { '\u2032', '\'' },
+            { '\u2033', '\"' }
+        };
+        foreach (var key in qmap.Keys)
+            if (buffer.IndexOf(key) > -1)
```
Comment (david-allison) [inline CorpusSearch/Extensions/NormalizationExtensions.cs:51]: Please use braces unless it's a guard clause

**Review by david-allison — COMMENTED** [review]
One nitpick, then good to go, thanks!

File: CorpusSearch/Extensions/NormalizationExtensions.cs (hunk 3 above)
Comment (lauterb) [reply CorpusSearch/Extensions/NormalizationExtensions.cs:51]: You mean for both the for and the if?

Comment (lauterb) [reply CorpusSearch/Extensions/NormalizationExtensions.cs:51]: I assume definitely for the if
And most likely for the for as well

Comment (lauterb) [reply CorpusSearch/Extensions/NormalizationExtensions.cs:51]: Note: cleanup in Rider  actually removes the braces

Comment (lauterb) [reply CorpusSearch/Extensions/NormalizationExtensions.cs:51]: Okay, if I just do reformat but no syntax style it will leave them

**Review by david-allison — APPROVED** [review]
Cheers!

**Pull request merged**